### PR TITLE
fix useUndo functions(count, history, clear)

### PR DESCRIPTION
**Pull request by Nerevar123**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
count() and history() throws "TypeError: Cannot read property '_history' of undefined",
clear() cleaning current tree

Issue Number: #66 

## What is the new behavior?
count() return number of possible undos,
history() return array of previous states,
clear(nodes, edges) clean history and make first element of new history with current nodes and edges

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/helpers/useUndo.ts [hunk 1]
```diff
@@ -162,7 +162,7 @@ export const useUndo = ({
     });
   }, []);
 
-  const clear = useCallback(() => {
+  const clear = useCallback((nodes, edges) => {
```
Comment (amcdnl) [inline src/helpers/useUndo.ts:165]: Needs types

File: src/helpers/useUndo.ts [hunk 2]
```diff
@@ -80,7 +80,7 @@ export interface UndoResult {
   /**
    * Clear state.
    */
-  clear: () => void;
+  clear: (nodes: NodeData[], edges: EdgeData[]) => void;
```
Comment (amcdnl) [inline src/helpers/useUndo.ts:83]: I guess this is technically a breaking api change?

**Review by amcdnl — CHANGES_REQUESTED** [review]

**Comment by amcdnl** [comment]
Need to update the markdown docs too

File: src/helpers/useUndo.ts (hunk 2 above)
Comment (Nerevar123) [reply src/helpers/useUndo.ts:83]: now clear() not working at all, just clean a whole tree.
so technically yes, but in practice, nobody could use it

**Comment by Nerevar123** [comment]
> Need to update the markdown docs too

done

**Comment by amcdnl** [comment]
released

**Pull request merged**